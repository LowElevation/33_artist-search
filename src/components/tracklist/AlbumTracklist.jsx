import React from "react";
import PropTypes from "prop-types";
import uuid from "react-uuid";
import { Link } from "react-router-dom";

const AlbumTracklist = ({ tracks, handleClick }) => {
  const tracksList = tracks.map(track => {
    return (
      <li key={uuid()}>
        <Link to={`/releases/tracklist/${track.trackID}`} onClick={handleClick}>
          {track.title}
        </Link>

      </li>
    );
  });
  return (
    <ul>
      {tracksList}
    </ul>
  );
};

AlbumTracklist.propTypes = {
  tracks: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired
  })),
  handleClick: PropTypes.func
};

export default AlbumTracklist;
