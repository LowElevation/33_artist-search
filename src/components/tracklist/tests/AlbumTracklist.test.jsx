import React from "react";
import { render, cleanup } from "@testing-library/react";
import AlbumTracklist from "../AlbumTracklist";

describe("AlbumTracklist component", () => {
  afterEach(() => cleanup());
  it("renders AlbumTracklist", () => {
    const { asFragment } = render(<AlbumTracklist tracks={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
