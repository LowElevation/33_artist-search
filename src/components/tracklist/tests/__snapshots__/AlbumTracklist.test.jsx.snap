// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumTracklist component renders AlbumTracklist 1`] = `
<DocumentFragment>
  <ul />
</DocumentFragment>
`;
